### DISCUSSION TITLE
How can I migrate all my data to a new server?

### DISCUSSION THREAD

**Discussion by IzMichael** [plausible/analytics#2557] [Self-Hosted Support]
Hiya, I'm in the process of migrating all of my data and apps to a new server, and I'm trying to do the same to my Plausible instance. Unfortunately, I can't figure out how I can get access to the database(s) and move them to the new server. If anyone can help, or at least point me in the right direction, that would be greatly appreciated!

**Comment by ruslandoga** [comment]
@IzMichael 👋

The data is split between postgres and clickhouse.

If you use [the docker compose setup,](https://github.com/plausible/hosting/blob/master/docker-compose.yml) you can use commands like `docker compose exec -ti plausible_db ash` and `docker compose exec -ti plausible_events_db ash` to access the running containers. From there you can follow [`pg_dump`](https://www.postgresql.org/docs/current/app-pgdump.html) and [clickhouse backup](https://clickhouse.com/docs/en/operations/backup/) guides to export your data into mounted volumes.

**Reply by ruslandoga** [reply]
[This postgres upgrade guide](https://github.com/plausible/hosting/blob/master/upgrade/postgres.md) could also be helpful as it describes data export/import steps in more detail. Similar approach could be used to export/import clickhouse data.

**Comment by infosion** [comment]
I am also in the situation of having to migrate to a new server. I am very reluctant to lose data from the statistics. 
Maybe someone has a step by step guide on how to best move both the Docker image and the databases to the new server?
